### PR TITLE
developer_guides: debugability: probes: Update for sof client version…

### PR DESCRIPTION
**Pull request by ujfalusi**
… of probes

The probes got converted to sof client mode, update the documentation to
be up to date.

Signed-off-by: Peter Ujfalusi <peter.ujfalusi@linux.intel.com>

### CODE REVIEW

**Review by plbossart — CHANGES_REQUESTED** [review]
typos or sticky fingers ;-)

File: developer_guides/debugability/probes/index.rst [hunk 1]
```diff
@@ -17,15 +17,49 @@ Requirements
...
+  The debugfs also needs to be enabled for the probes to be usable.
 
   .. code-block:: bash
 
      CONFIG_DEBUG_FS=y
-     CONFIG_SND_SOC_SOF_DEBUG_PROBES=y
-     CONFIG_SND_SOC_SOF_HDA_PROBES=y
 
-- Enable the Probe module in the SOF firmware ``kconfig`` using this command:
+- The probes client needs to be enabled via module parameter (e.g. ``/etc/modprobe.d/sof.conf``):
```
[the remark this answers is not in the document]
Comment (ujfalusi) [reply developer_guides/debugability/probes/index.rst:31]: fixing all of these...

**Comment by ujfalusi** [comment]
Changes since v1:
- spelling fixes pointed out by @plbossart

**Review by plbossart — APPROVED** [review]
Thanks @ujfalusi

**Review by kv2019i — APPROVED** [review]
Thanks @ujfalusi !

**Pull request merged**